test/gateway/eip20_gateway/confirm_redeem_intent.js

```diff
@@ -0,0 +1,494 @@
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const EIP20Gateway = artifacts.require('TestEIP20Gateway.sol'),
+  Token = artifacts.require("MockUtilityToken"),
+  Utils = require('../../test_lib/utils'),
+  MockToken = artifacts.require("MockToken"),
+  BN = require('bn.js'),
+  TestDataJSON = require('./test_data/redeem_progressed_1'),
+  TestDataJSON2 = require('./test_data/redeem_progressed_2'),
+  messageBus = require('../../test_lib/message_bus.js'),
+  MockOrganization = artifacts.require('MockOrganization.sol'),
+  EventDecoder = require('../../test_lib/event_decoder.js');
+
+let MessageStatusEnum = messageBus.MessageStatusEnum;
+
+contract('EIP20Gateway.confirmRedeemIntent() ', function (accounts) {
+
+  let utilityToken,
+    bountyAmount,
+    coreAddress,
+    owner,
+    burnerAddress,
+    dummyStateRootProvider,
+    eip20Gateway,
+    mockToken,
+    baseToken,
+    mockOrganization,
+    worker,
+    redeemRequest,
+    redeemRequestWithNonceTwo,
+    messageHash = TestDataJSON.gateway.confirm_redeem_intent.return_value.returned_value.messageHash_;
+
+  async function confirmRedeemIntent(stubData) {
+
+    await eip20Gateway.confirmRedeemIntent(
+      stubData.redeemer,
+      stubData.nonce,
+      stubData.beneficiary,
+      stubData.amount,
+      stubData.gasPrice,
+      stubData.gasLimit,
+      stubData.blockNumber,
+      stubData.hashLock,
+      stubData.storageProof,
+    );
+  }
+
+  beforeEach(async function () {
+
+      mockToken = await MockToken.new({from: accounts[0]});
+      baseToken = await MockToken.new({from: accounts[0]});
+
+      redeemRequest = TestDataJSON.gateway.confirm_redeem_intent.params;
+      redeemRequest.nonce = new BN(redeemRequest.nonce, 16);
+      redeemRequest.amount = new BN(redeemRequest.amount, 16);
+      redeemRequest.gasLimit = new BN(redeemRequest.gasLimit, 16);
+      redeemRequest.blockNumber = new BN(redeemRequest.blockNumber, 16);
+      redeemRequest.gasPrice = new BN(redeemRequest.gasPrice, 16);
+
+      redeemRequestWithNonceTwo = TestDataJSON2.gateway.confirm_redeem_intent.params;
+      redeemRequestWithNonceTwo.nonce = new BN(redeemRequestWithNonceTwo.nonce, 16);
+      redeemRequestWithNonceTwo.amount = new BN(redeemRequestWithNonceTwo.amount, 16);
+      redeemRequestWithNonceTwo.gasLimit = new BN(redeemRequestWithNonceTwo.gasLimit, 16);
+      redeemRequestWithNonceTwo.blockNumber = new BN(redeemRequestWithNonceTwo.blockNumber, 16);
+      redeemRequestWithNonceTwo.gasPrice = new BN(redeemRequestWithNonceTwo.gasPrice, 16);
+
+      owner = accounts[4];
+      worker = accounts[8];
+
+      mockOrganization = await MockOrganization.new(owner, worker);
+
+      let symbol = "OST",
+        name = "Simple Token",
+        decimals = 18;
+
+      utilityToken = await Token.new(
+        accounts[3],
+        symbol,
+        name,
+        decimals,
+        owner,
+        {from: accounts[0]},
```

Comment:
Please check if this is required and used in any of the tests? If not then please remove it.

```
let symbol = "OST",
        name = "Simple Token",
        decimals = 18;

      utilityToken = await Token.new(
        accounts[3],
        symbol,
        name,
        decimals,
        owner,
        {from: accounts[0]},
      );
```